### DISCUSSION TITLE
Spatial coordinates based evaluation

### DISCUSSION THREAD

**Discussion by somsubhro88** [nschloe/meshio#1214] [Q&A] [answered]
Hello, @nschloe and others,

First of all, thank you for the effort in making this library. I have a specific question to my application.

I have a .vtu file which has a mesh and associated data. Below is an image of what the mesh looks like - 
![image](https://user-images.githubusercontent.com/20040210/141322782-dc4f4611-f195-42a2-93ef-d6eb82249ed9.png)

Below is an the same mesh with "psi" , the data in the mesh I would like to work with - 
![image](https://user-images.githubusercontent.com/20040210/141322986-b3c7ed5a-46f0-4021-87a2-281906809c3a.png)

What I would like to do is explained in the image below - I would like to extract a list of "psi" values along the y-axis. There is a long version of solving this - extracting each value separately and then matching them up. I was wondering if there's clever way of doing this using already defined functions? My primary goal is to be able to measure the circular feature that you see in the images, and my criteria for measuring is "psi".
![image](https://user-images.githubusercontent.com/20040210/141323462-3e5124af-8894-4fad-b851-212af63db8a3.png)

Any help or guidance would be greatly appreciated! I have also attached the .vtu file.
[solution-00100000.zip](https://github.com/nschloe/meshio/files/7521463/solution-00100000.zip)

Again, thank you!
Som

**Comment by nschloe** [comment]
[First of all, don't use the jet color scheme.](https://blogs.mathworks.com/headlines/2018/10/10/a-dangerous-rainbow-why-colormaps-matter/) You'll see that your circle doesn't have a boundary as sharp as it looks in the images.

About your problem: Evaluating functions on meshes can be hard. The main issue is finding in which cell a particular point (x,y) sits. I wouldn't know how to do this except one-by-one.

**Comment by keurfonluu** [comment, marked as answer]
`meshio` is simply a mesh reader/writer library. To achieve what you want to do, you need more advanced tools.

You can check out `pyvista`'s function [`sample_over_line`](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSet.sample_over_line.html?highlight=sample%20over%20line#pyvista.DataSet.sample_over_line) to extract data along a line. `pyvista` is a wrapper for VTK which partially relies on `meshio` too for mesh I/O.

**Comment by somsubhro88** [comment]
Thank you, @nschloe and @keurfonluu .

I shall explore both your recommendations, and will update this post once I've found a concrete solution.
